libs/ledger-live-common/src/api/Manager.ts

```diff
@@ -571,6 +571,28 @@ const installMcu = (
   }).pipe(remapSocketError(context));
 };
 
+function retrieveMcuVersion(
+  finalFirmware: FinalFirmware
+): Promise<McuVersion | undefined> {
+  return getMcus()
+    .then((mcus) =>
+      mcus.filter((deviceInfo) => {
+        const provider = getProviderId(deviceInfo);
+        return (mcu) => mcu.providers.includes(provider);
+      })
+    )
+    .then((mcus) =>
+      mcus.filter((mcu) => mcu.from_bootloader_version !== "none")
+    )
+    .then((mcus) =>
+      findBestMCU(
+        finalFirmware.mcu_versions
+          .map((id) => mcus.find((mcu) => mcu.id === id))
```

Comment:
what happens if `mcus` is undefined/null (i don't think it's possible, `filter` returns an empty array if no mcu is found) - but what about `getMcus`, if it fails or returns `undefined` ?

Probably not the correct PR to do this - but i don't like this Manager 🙃

Reply:
getMcus uses some cache mechanism inside libs/coin-framework, I believe it handles the promise failing correctly, because it's nowhere used with a try catch. This whole manager is probably something worth rethinking.

Reply:
ahah yeah, if to `getMcus` we need something that is located inside `libs/coin-framework` we need to re-work it ;) (but not in this PR)